lib/dal/src/workspace_snapshot/graph.rs

```diff
@@ -1013,45 +1014,63 @@ impl WorkspaceSnapshotGraph {
             let to_rebase_item_weight =
                 self.get_node_weight(only_to_rebase_edge_info.target_node_index)?;
 
-            if to_rebase_edge_weight
+            // This is an edge that is only to_rebase. So either:
+            // -- Onto has seen this edge:
+            //     -- So, if to_rebase has modified the target of the edge since onto last saw the target,
+            //      we should produce a ModifyRemovedItem conflict,
+            //     -- OR, if to_rebase has *not* modified the target, we should produce a RemoveEdge update.
+            // -- Onto has never seen this edge:
+            //     -- So, either we should just silently let the edge stay in to_rebase, OR, we should produce
+            //          an exclusive edge conflict
```

Comment:
Worth pointing out that the reason this scenario would happen is if the edge were added to `to_rebase` **after** `onto` was forked from `to_rebase`?